src/lib/archives.c

```diff
@@ -225,28 +225,47 @@ int archives_check_single_file_tarball(const char *tarfilename, const char *file
 {
 	struct archive *a;
 	struct archive_entry *entry;
-	int r;
-	bool found = false;
+	const char *file_entry;
+	size_t file_len;
+	int ret = -ENOENT;
+
+	// Read archive
+	ret = archive_from_filename(&a, tarfilename);
+	if (ret < 0) {
+		return ret;
+	}
 
-	/* set up read */
-	r = archive_from_filename(&a, tarfilename);
-	if (r < 0) {
-		return r;
+	ret = -ENOENT;
+	file_len = strlen(file);
```

Comment:
The `file_len` value here appears to be unused, since it is soon overwritten on line 249.

Reply:
It's a left over. Thanks

Reply:
And I had the same problem with ret. I fixed that now.